executor/runtime/docker/docker.go

```diff
@@ -1869,6 +1870,37 @@ func (r *DockerRuntime) getPlaformContainerNames() []string {
 	return platformContainerNames
 }
 
+func (r *DockerRuntime) getContainerVolumeMounts(mainContainerRoot string, c v1.Container, pod v1.Pod) []mount.Mount {
+	mounts := []mount.Mount{}
+	volumes := pod.Spec.Volumes
+	for _, volumeMount := range c.VolumeMounts {
+		v := getVolumeByName(volumes, volumeMount.Name)
+		if v.Name == "" {
+			continue
+		}
+		if v.FlexVolume.Driver == "TitusMainContainerVolume" {
+			m := mount.Mount{
+				Type:        "bind",
+				Source:      mainContainerRoot + v.FlexVolume.Options["source"],
+				Target:      volumeMount.MountPath,
+				ReadOnly:    volumeMount.ReadOnly,
+				BindOptions: &mount.BindOptions{Propagation: mount.PropagationRShared},
+			}
+			mounts = append(mounts, m)
+		}
+	}
+	return mounts
+}
+
+func getVolumeByName(volumes []v1.Volume, name string) v1.Volume {
+	for _, v := range volumes {
```

Comment:
It might be easier to return *v1.Volume here.

Reply:
Why though?